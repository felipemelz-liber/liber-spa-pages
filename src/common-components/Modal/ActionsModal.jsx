import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'liber-components/components/Dialog';
import { SuccessButton, PrimaryButton, AlertButton, ErrorButton } from 'liber-components';
import LoadingButton from '../Buttons/LoadingButton';
import {
  Button,
  DialogContent,
  DialogHeader,
  DialogActions,
  Title,
  IconButton,
  CloseIcon,
} from './ActionsModal.styles';

const ActionsModal = ({ title, children, show, onClose, confirmationButtonProps, closable }) => {
  const renderConfirmationButton = () => {
    const { onConfirmation, disabled, type, text, loading } = confirmationButtonProps;
    const isLoadingButton = loading !== null && loading !== undefined;
    const buttonProps = {
      onClick: onConfirmation,
      disabled,
      children: text ? String(text).toUpperCase() : 'CONFIRMAR',
      version: 2,
    };
    switch (type) {
      case 'primary':
        return isLoadingButton ? (
          <LoadingButton ButtonComponent={PrimaryButton} loading={loading} {...buttonProps} />
        ) : (
          <PrimaryButton {...buttonProps} />
        );
      case 'alert':
        return isLoadingButton ? (
          <LoadingButton ButtonComponent={AlertButton} loading={loading} {...buttonProps} />
        ) : (
          <AlertButton {...buttonProps} />
        );
      case 'error':
        return isLoadingButton ? (
          <LoadingButton ButtonComponent={ErrorButton} loading={loading} {...buttonProps} />
        ) : (
          <ErrorButton {...buttonProps} />
        );
      default:
        return isLoadingButton ? (
          <LoadingButton ButtonComponent={SuccessButton} loading={loading} {...buttonProps} />
        ) : (
          <SuccessButton {...buttonProps} />
        );
    }
  };

  return (
    <Modal show={show} onLeaved={onClose} closable={closable}>
      <DialogHeader>
        <Title>{title}</Title>
        <IconButton size="large" circle onClick={closable ? onClose : undefined} version={2}>
          <CloseIcon />
        </IconButton>
      </DialogHeader>
      <DialogContent>{children}</DialogContent>
      <DialogActions>
        <Button onClick={closable ? onClose : undefined} version={2}>
          CANCELAR
        </Button>
        {renderConfirmationButton()}
      </DialogActions>
    </Modal>
  );
};

ActionsModal.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  show: PropTypes.bool,
  onClose: PropTypes.func,
  closable: PropTypes.bool,
  confirmationButtonProps: PropTypes.shape({
    onConfirmation: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.oneOf(['primary', 'success', 'alert', 'error']),
    text: PropTypes.string,
    loading: PropTypes.bool,
  }),
};

ActionsModal.defaultProps = {
  title: 'Confirmação',
  children: <div />,
  show: true,
  closable: true,
  onClose: () => null,
  confirmationButtonProps: {
    onConfirmation: () => null,
    disabled: false,
    type: 'success',
    text: 'Confirmar',
    loading: null,
  },
};

export default ActionsModal;
