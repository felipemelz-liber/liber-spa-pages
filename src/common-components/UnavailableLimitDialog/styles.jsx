import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
