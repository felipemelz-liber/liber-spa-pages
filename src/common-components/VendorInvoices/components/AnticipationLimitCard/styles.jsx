import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
