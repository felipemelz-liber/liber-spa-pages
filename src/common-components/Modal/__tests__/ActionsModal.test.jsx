import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { SuccessButton, PrimaryButton, AlertButton, ErrorButton } from 'liber-components';
import { Button, IconButton } from '../ActionsModal.styles';
import ActionsModal from '../ActionsModal';
import LoadingButton from '../../Buttons/LoadingButton';

describe('ActionsModal component tests', () => {
  configure({ adapter: new Adapter() });

  const buttonTypes = ['success', 'primary', 'alert', 'error'];
  const props = {
    title: 'TÃ­tulo novo',
    children: <div>Mock</div>,
    show: true,
    onClose: sinon.spy(),
    confirmationButtonProps: {
      onConfirmation: sinon.spy(),
      disabled: false,
      text: 'Confirmar novo',
    },
  };

  afterEach(() => {
    const {
      onClose,
      confirmationButtonProps: { onConfirmation },
    } = props;
    onClose.resetHistory();
    onConfirmation.resetHistory();
  });

  it('should render ActionsModal correctly without props', () => {
    const wrapper = shallow(<ActionsModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render ActionsModal correctly with props', () => {
    const wrapper = shallow(<ActionsModal {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  buttonTypes.forEach(type => {
    it(`should render ActionsModal correctly with ${type} button type`, () => {
      const { confirmationButtonProps } = props;
      const wrapper = shallow(
        <ActionsModal {...props} confirmationButtonProps={{ ...confirmationButtonProps, type }} />,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  it('should call onClose when clicking on cancel button', () => {
    const wrapper = shallow(<ActionsModal {...props} />);
    const cancelButton = wrapper.find(Button);
    const { onClose } = props;
    cancelButton.simulate('click');
    expect(onClose.called).toBeTruthy();
  });
  it('should call onClose when clicking on close button', () => {
    const wrapper = shallow(<ActionsModal {...props} />);
    const closeButton = wrapper.find(IconButton);
    const { onClose } = props;
    closeButton.simulate('click');
    expect(onClose.called).toBeTruthy();
  });
  [SuccessButton, PrimaryButton, AlertButton, ErrorButton].forEach((ButtonComponent, index) => {
    it(`should call onConfirmation when clicking on ${buttonTypes[index]} button`, () => {
      const { confirmationButtonProps } = props;
      const wrapper = shallow(
        <ActionsModal
          {...props}
          confirmationButtonProps={{ ...confirmationButtonProps, type: buttonTypes[index] }}
        />,
      );
      const confirmationButton = wrapper.find(ButtonComponent);
      confirmationButton.simulate('click');
      expect(confirmationButtonProps.onConfirmation.called).toBeTruthy();
    });
  });
  it('should have a disabled confirmation button when disabled prop is true on confirmationButtonProps', () => {
    const { confirmationButtonProps } = props;
    const wrapper = shallow(
      <ActionsModal
        {...props}
        confirmationButtonProps={{ ...confirmationButtonProps, disabled: true }}
      />,
    );
    const confirmationButton = wrapper.find(SuccessButton);
    expect(confirmationButton.prop('disabled')).toBeTruthy();
  });
  it('should have a LoadingButton when loading prop is provided on confirmationButtonProps', () => {
    const { confirmationButtonProps } = props;
    const wrapper = shallow(
      <ActionsModal
        {...props}
        confirmationButtonProps={{ ...confirmationButtonProps, loading: false }}
      />,
    );
    const loadingButton = wrapper.find(LoadingButton);
    expect(loadingButton.length).toBeGreaterThan(0);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
