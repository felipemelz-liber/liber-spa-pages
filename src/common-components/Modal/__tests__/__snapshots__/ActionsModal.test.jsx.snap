// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsModal component tests should have a LoadingButton when loading prop is provided on confirmationButtonProps 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <LoadingButton
      ButtonComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Buttons__SuccessButton-u35aut-3",
            "isStatic": false,
            "rules": Array [],
          },
          "displayName": "Buttons__SuccessButton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Buttons__SuccessButton-u35aut-3",
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      disabled={false}
      loading={false}
      onClick={[Function]}
      size="medium"
      spinnerColor="white"
      version={2}
    >
      CONFIRMAR NOVO
    </LoadingButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly with alert button type 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__AlertButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR NOVO
    </Buttons__AlertButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly with error button type 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__ErrorButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR NOVO
    </Buttons__ErrorButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly with primary button type 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__PrimaryButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR NOVO
    </Buttons__PrimaryButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly with props 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__SuccessButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR NOVO
    </Buttons__SuccessButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly with success button type 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Título novo
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div>
      Mock
    </div>
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__SuccessButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR NOVO
    </Buttons__SuccessButton>
  </styled.div>
</Modal>
`;

exports[`ActionsModal component tests should render ActionsModal correctly without props 1`] = `
<Modal
  animateFrom="-150px"
  animation={
    Object {
      "in": "cubic-bezier(.17,1.92,.3,.71)",
      "out": "cubic-bezier(1,.23,.76,-1.11)",
    }
  }
  closable={true}
  closeButton={false}
  closeButtonPadding="10px 14px"
  duration={270}
  elementId="modal"
  fullscreen={false}
  insertComponent={null}
  label=""
  labelAlign="left"
  onAppearStart={[Function]}
  onAppeared={[Function]}
  onLeaveStart={[Function]}
  onLeaved={[Function]}
  overflow=""
  position={null}
  show={true}
>
  <styled.div>
    <styled.span>
      Confirmação
    </styled.span>
    <Styled(Buttons__Button)
      circle={true}
      onClick={[Function]}
      size="large"
      version={2}
    >
      <Styled(MdClose) />
    </Styled(Buttons__Button)>
  </styled.div>
  <styled.div>
    <div />
  </styled.div>
  <styled.div>
    <Styled(Buttons__Button)
      onClick={[Function]}
      version={2}
    >
      CANCELAR
    </Styled(Buttons__Button)>
    <Buttons__SuccessButton
      disabled={false}
      onClick={[Function]}
      version={2}
    >
      CONFIRMAR
    </Buttons__SuccessButton>
  </styled.div>
</Modal>
`;
